projects/hmcts-annotation-ui-lib/src/lib/data/pdf-adapter.spec.ts

```diff
@@ -0,0 +1,407 @@
+import { TestBed, inject } from '@angular/core/testing';
+import { Utils } from './utils';
+import { PdfAdapter } from './pdf-adapter';
+import { AnnotationSet, Annotation, Comment, Rectangle } from './annotation-set.model';
+import { WINDOW } from '@ng-toolkit/universal';
+import { EmLoggerService } from '../logging/em-logger.service';
+
+class MockUtils {
+    generateRectanglePerLine() {}
+}
+
+class MockWindow {
+    getSelection() {
+        return null;
+    }
+}
+
+describe('PdfAdapter', () => {
+    const mockUtils = new MockUtils();
+    const mockRectangle = new Rectangle('63225ccd-61fe-4aa7-8c5f-cf9bc31cc424',
+        '4bcc2edf-487d-4ee0-a5b0-a3cdfe93bf1a',
+        '123141', null, new Date(), null, null, null,
+        9.6, 60,
+        50, 87);
+    const mockComment = new Comment(
+        'cfe6bdad-8fc5-4240-adfc-d583bdaee47a',
+        '22a3bde9-18d6-46b2-982b-36e0a631ea4b',
+        '111111', null, new Date(), null, null, null,
+        'comment content');
+    const mockAnnotation = new Annotation(
+        '22a3bde9-18d6-46b2-982b-36e0a631ea4b',
+        '9ad31e66-ec05-476d-9a38-09973d51c0c3',
+        '111111', new Date(), null,
+        '111111', null, new Date(), 'docId',
+        1, 'red', [mockComment], [mockRectangle], 'highlight'
+    );
+    const mockAnnotationSet = new AnnotationSet(
+        '9ad31e66-ec05-476d-9a38-09973d51c0c3',
+        '111111',
+        null,
+        new Date(),
+        '111111', null,
+        new Date(),
+        '',
+        [mockAnnotation]
+    );
+    const mockWindow = new MockWindow();
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+          providers: [
+                EmLoggerService,
+                PdfAdapter,
+                { provide: WINDOW, useFactory: () => mockWindow},
```

Comment:
space, the final frontier, on the right

Reply:
tabs vs spaces. the battle with no winners

Reply:
lool, i meant spacing on the curly brackets at the end just like the beginning one has